openedx/openedx-platform
# Third party auth: Support for running behind a reverse proxy on a non-standard port

**Desciprion:** This PR adds support for `HTTP_X_FORWARDED_HOST` and `HTTP_X_FORWARDED_PORT` headers for means of third party authentication.
**Background:** If LMS runs on a non-standard http port (i.e. not 80), `python-social-auth` detects that and fails authentication with message "Request was sent to lms_hostname:actual_lms_port instead of lms_hostname". Common way to pass actual host and port to proxied server is to set `HTTP_X_FORWARDED_HOST` and `HTTP_X_FORWARDED_PORT` headers. This PR makes python-social-auth prefer values of these headers, if present, over values in `request.META['SERVER_PORT`]`and`request.get_host`
**Related:** [Solutions fork PR](https://github.com/edx-solutions/edx-platform/pull/530)
**JIRA Ticket:** https://openedx.atlassian.net/browse/YONK-83
**Sandbox:** [LMS](http://pr9848.sandbox.opencraft.com/), [Studio](http://studio.pr9848.sandbox.opencraft.com/)
**Testing instructions:**
1. Open Login page in sandbox' LMS
2. Click "Sign in with TestShib" - you'll be redirected to testshib login page
3. Log in to testshib as one of the provided users (myself, alterego or superego - passwords same as username)
4. You should be redirected to LMS login page with message "You've successfully logged into your TestShib account, but this account isn't linked with an edx-platform account".

_Test failure conditions:_ at step 4, error message "Authentication failed: SAML login failed: ['invalid_response'] The response was received at http://pr9848.sandbox.opencraft.com:8000/auth/complete/tpa-saml/ instead of http://pr9848.sandbox.opencraft.com/auth/complete/tpa-saml/" appears

---

**Settings**

``` yaml
SOCIAL_AUTH_RESPECT_X_FORWARDED_HEADERS: true
EDXAPP_FEATURES:
  ENABLE_THIRD_PARTY_AUTH: true
```
